build-common/common.xml

```diff
@@ -11,15 +11,26 @@
 	<!-- Path to the folder that contain the external assemblies -->
 	<property name="lib.dir" value="lib" dynamic="true" />
 
-	<!-- This is used only for build folder -->
-	<!-- TODO: Either remove or refactor to use NHibernate.props -->
-	<property name="project.version" value="5.2.7" overwrite="false" />
-	<property name="project.version.numeric" value="5.2.7" overwrite="false" />
+  <!-- This is used for build folder and also for "API Reference" docs (not active task, see doc/NHibernate.shfbproj.template) -->
+  <target name="get-project-version">
+    <xmlpeek
+      file="${root.dir}/build-common/NHibernate.props"
+      xpath="/Project/PropertyGroup/VersionPrefix"
+      property="version.prefix" />
+    <property name="project.version" value="${version.prefix}" />
+  </target>
+  
+  <target name="set-doc-version">
+    <xmlpoke
+      file="${root.dir}/doc/reference/master.xml"
```

Comment:
This causes the git "workspace" to get "dirty", as this file is not ignored by git. Could you add so final steps to reference build file targets in order to clean it after builds?

Reply:
git ignore wouldn't help here  as it only works for new files. Maybe we should rename this file (`common.xml.template?`) and copy it after modification? That way common.xml can be added to git ignore

Reply:
> Maybe we should rename this file (common.xml.template?) and copy it after modification? That way common.xml can be added to git ignore

The altered file is `doc/reference/master.xml`, so that is not `common.xml` which should have an ignored version with a template, but `master.xml`.
And I propose another solution : add a step after the build to change back the file to its initial value, avoiding it to be considered altered.

Reply:
I decided to go with additional file:
1) To keep all logic in one step;
2) To avoid dirty workspace in case of errors or build interruption;